tests/test_web/test_api_v1.py

```diff
@@ -206,13 +206,124 @@ def test_add_bundle_success(db, auth_env, client):
         'state_history': [{
             'state': 'in_progress',
             'state_reason': 'The request was initiated',
-            'updated': datetime.datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+            'updated': '2020-02-12T17:03:00Z'
         }],
         'state_reason': 'The request was initiated',
-        'updated': datetime.datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ'),
+        'updated': '2020-02-12T17:03:00Z',
         'user': 'tbrady@DOMAIN.LOCAL'
     }
 
     rv = client.post('/api/v1/builds/add', json=data, environ_base=auth_env)
+    rv_json = rv.json
+    rv_json['state_history'][0]['updated'] = '2020-02-12T17:03:00Z'
+    rv_json['updated'] = '2020-02-12T17:03:00Z'
     assert rv.status_code == 201
-    assert response_json == rv.json
+    assert response_json == rv_json
+
+
+@pytest.mark.parametrize(
+    'arches, binary_image_resolved, from_index_resolved, index_image, state, state_reason,'
+    'error_msg', (
+        ([''],
+         'resolved:binary',
+         'resolved:index',
+         'index:image',
+         'state',
+         'state:reason',
+         'Architectures should be specified as a non-empty array of strings'
+         ),
+        (['s390x'],
+         '',
+         'resolved:index',
+         'index:image',
+         'state',
+         'state:reason',
+         'The value for "binary_image_resolved" must be a non-empty string'
+         ),
+        (['s390x'],
+         'resolved:binary',
+         'resolved:index',
+         'index:image',
+         'no_state',
+         'state_reason',
+         'The "state" key is required when "state_reason" is supplied'
+         ),
+    ))
+def test_patch_request_invalid_params_format(
+    arches, binary_image_resolved, from_index_resolved, index_image,
+    state, state_reason, error_msg, db, worker_auth_env, client
+):
+    data = {
+        'arches': arches,
+        'binary_image_resolved': binary_image_resolved,
+        'from_index_resolved': from_index_resolved,
+        'index_image': index_image,
+        'state': state,
+        'state_reason': state_reason
+    }
+
+    if data['state'] == 'no_state':
+        data.pop('state')
+    binary_image = models.Image(pull_specification='quay.io/image:latest')
+    db.session.add(binary_image)
+    request = models.Request(
+        binary_image=binary_image,
+        type=models.RequestTypeMapping.add.value,
+    )
+    db.session.add(request)
+
+    rv = client.patch('/api/v1/builds/1', json=data, environ_base=worker_auth_env)
+    assert rv.status_code == 400
+    assert error_msg == rv.json['error']
+
+
+def test_patch_request_success(db, worker_auth_env, client):
+    data = {
+        'arches': ['arches'],
+        'state': 'complete',
+        'state_reason': 'All done!',
+        'index_image': 'index:image'
+    }
+
+    response_json = {
```

Comment:
Since we already test the JSON method elsewhere, you could just do:
```python
assert rv_json == db.session.query(Request).get(1).to_json(verbose=True)
```

Reply:
I prefer having the entire json in the success test. It helps get a glance of the response without having to navigate out of your editor. If you feel strongly against it I can change it.